lib/monitoring/aws-docdb/DocumentDbMetricFactory.ts

```diff
@@ -0,0 +1,121 @@
+import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";
+import { DatabaseCluster } from "aws-cdk-lib/aws-docdb";
+
+import {
+  getLatencyTypeLabel,
+  getLatencyTypeStatistic,
+  LatencyType,
+  MetricFactory,
+  MetricStatistic,
+} from "../../common";
+
+const DocumentDbNamespace = "AWS/DocDB";
+
+export interface DocumentDbMetricFactoryProps {
+  /**
+   * database cluster identifier (either this or `cluster` need to be specified)
+   */
+  readonly clusterIdentifier?: string;
```

Comment:
Do we really need to support both? We typically are not doing that for other monitorings.

Reply:
I know, but I was thinking about two reasons:

1) The cluster L2 constructs are quite new and users might not be using them yet (they might only have cluster ID available)
2) The other construct (RDS) was using identifier and should be using cluster too - here I wanted to be backwards compatible and consistent

Not sure it is the best decision, what do you think?

Reply:
1. Even though it is new, if it comes under our minimum supported `aws-cdk-lib` version we should use it. For the users who might have just the ID available, they can import the Cluster via [fromDatabaseClusterAttributes](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_docdb.DatabaseCluster.html#static-fromwbrdatabasewbrclusterwbrattributesscope-id-attrs)  which takes [DatabaseClusterAttributes](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_docdb.DatabaseClusterAttributes.html) which inturn only needs the ID as required argument.
2. For RDS, we can keep it the way you have right now (supporting both) but we should mark the ID as depreacated and instruct them to use the IDatabaseCluster instead (same things apply here)

Reply:
OK, no problem, I can do it like this too.